msamp/common/tensor/cast.py

```diff
@@ -77,22 +77,20 @@ def cast_to_fp16(input, meta, sync=False):
             torch.Tensor: tensor whose dtype is torch.float16.
         """
         meta.amax[0] = input.abs().max()
-        in_time = meta.is_in_time_scaling()
         if sync:
             # convert NAN to INF since NCCL-ReduceMax ignores NAN
             # notice: nan and posinf must be INF
             meta.amax[0].nan_to_num_(nan=torch.inf, posinf=torch.inf)
             world_size = DistUtil.get_world_size()
             if world_size > 1:
                 dist.all_reduce(meta.amax[0], op=dist.ReduceOp.MAX, group=meta.group)
-        if in_time or sync:
-            # notice: we scale the tensor with qtype FP8-E4M3.
-            meta.reset_scaling_factor(qtype=Dtypes.kfloat8_e4m3)
-        meta.scale.clamp_(max=Floating.qfp_max[meta.qtype])
+        meta.reset_scaling_factor()
 
         meta.scale_inv.data.copy_(torch.reciprocal(meta.scale))    # scale_inv = 1 / scale
-        input_fp16 = (input * meta.scale).to(torch.float16)
-        return input_fp16
+        dtype = Dtypes.get_dtype_from_qtype(meta.qtype)
```

Comment:
This code change is for supporting bf16?

Reply:
Yes. It supports bf16/fp16 and fp32.